internal/framework/provider/provider.go

```diff
@@ -0,0 +1,220 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/provider/schema"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/framework/utils"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+
+	"github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+const (
+	endPointSTSDefault                    = "https://sts.amazonaws.com"
+	DeprecationMessage                    = "this resource is deprecated and will be removed in %s, please transition to %s"
+	DeprecationMessageParameterToResource = "this parameter is deprecated and will be removed in %s, please transition to %s"
+)
+
+var _ provider.Provider = (*MongodbtlasProvider)(nil)
+
+type MongodbtlasProvider struct {
+}
+
+type MongodbtlasProviderModel struct {
+	PublicKey         types.String `tfsdk:"public_key"`
+	PrivateKey        types.String `tfsdk:"private_key"`
+	BaseURL           types.String `tfsdk:"base_url"`
+	RealmBaseURL      types.String `tfsdk:"realm_base_url"`
+	IsMongodbGovCloud types.Bool   `tfsdk:"is_mongodbgov_cloud"`
+
+	// AssumeRole types.Object `tfsdk:"assume_role"`
+	SecretName           types.String `tfsdk:"secret_name"`
+	Region               types.String `tfsdk:"region"`
+	StsEndpoint          types.String `tfsdk:"sts_endpoint"`
+	AwsAccessKeyID       types.String `tfsdk:"aws_access_key_id"`
+	AwsSecretAccessKeyID types.String `tfsdk:"aws_secret_access_key"`
+	AwsSessionToken      types.String `tfsdk:"aws_session_token"`
+}
+
+type SecretData struct {
+	PublicKey  string `json:"public_key"`
+	PrivateKey string `json:"private_key"`
+}
+
+func (p *MongodbtlasProvider) Metadata(ctx context.Context, req provider.MetadataRequest, resp *provider.MetadataResponse) {
+	resp.TypeName = "mongodbatlas"
+}
+
+func (p *MongodbtlasProvider) Schema(ctx context.Context, req provider.SchemaRequest, resp *provider.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"public_key": schema.StringAttribute{
+				Optional:    true,
```

Comment:
required?

Reply:
These can't be required anymore since we don't have a DefaultFunc in the schema anymore. Hence, these need to be optional in the Schem() and need to be checked to be required in the Configure() method instead